packages/fxa-settings/src/lib/gql.test.ts

```diff
@@ -74,4 +77,26 @@ describe('errorHandler', () => {
 
     expect(window.location.replace).not.toHaveBeenCalled();
   });
+
+  it('does not redirect if the current page does not require the user to be authenticated', () => {
+    delete window.location;
+    window.location = {
+      replace: jest.fn(),
+      search: '',
+      pathname: 'foo',
+      href: pageWhichDoesNotRequireAuthentication,
+    };
```

Comment:
Not sure if this is relevant or helpful, but my eyes remembered seeing a recent tweak to this in @vbudhram's PR https://github.com/mozilla/fxa/pull/15057/files#diff-2c066da468bbb7a6e4b4512dd3ca2c46bcc1ef5ba04eb8b2b32ca134dddad8bf

Reply:
ah it must have gotten muddled in the rebase